src/chem/aeronet_aaod2ioda.py

```diff
@@ -0,0 +1,221 @@
+    f3_ssa = f3_ssa_all[['single_scattering_albedo[440nm]', 'single_scattering_albedo[675nm]',
+                         'single_scattering_albedo[870nm]', 'single_scattering_albedo[1020nm]']]
+    f3_tab = f3_tab_all[['absorption_aod[440nm]', 'absorption_aod[675nm]',
+                         'absorption_aod[870nm]', 'absorption_aod[1020nm]']]
+    f3 = pd.concat([f3_cad, f3_ssa, f3_tab], axis=1, join='inner')
+
+    # Define wavelengths, channels and frequencies of AERONET inversion data
+    aeronetinv_wav = np.array([440., 675, 870., 1020.], dtype=np.float32)
+    aeronetinv_chan = np.array([3, 5, 6, 7], dtype=np.intc)
+    speed_light = 2.99792458E8
+    frequency = speed_light*1.0E9/aeronetinv_wav
+    print('Output AERONET inverion data at wavelengths/channels/frequencies: ')
+    print(aeronetinv_wav)
+    print(aeronetinv_chan)
+    print(frequency)
+
+    nlocs, columns = f3.shape
+    if nlocs == 0:
+        print('No AERONET inversion data available in input files')
+        exit(0)
+
+    locationKeyList = [("latitude", "float"), ("longitude", "float"), ("datetime", "string"),]
+    writer = iconv.NcWriter(outfile, locationKeyList)
+    varDict = defaultdict(lambda: defaultdict(dict))
+    outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+    loc_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+    var_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+    units = {}
+    units['latitude'] = 'degree'
+    units['longitude'] = 'degree'
+    units['station_elevation'] = 'm'
+
+    # Define AERONET inversion varname that match with those in f3
+    # (e.g., match aaeronetinv_wav and aeronetinv_chan)
+    obsvars = {'aerosol_optical_depth_3': 'aod_coincident_input[440nm]',
+               'aerosol_optical_depth_5': 'aod_coincident_input[675nm]',
+               'aerosol_optical_depth_6': 'aod_coincident_input[870nm]',
+               'aerosol_optical_depth_7': 'aod_coincident_input[1020nm]',
+               'single_scattering_albedo_3': 'single_scattering_albedo[440nm]',
+               'single_scattering_albedo_5': 'single_scattering_albedo[675nm]',
+               'single_scattering_albedo_6': 'single_scattering_albedo[870nm]',
+               'single_scattering_albedo_7': 'single_scattering_albedo[1020nm]',
+               'absorption_aerosol_optical_depth_3': 'absorption_aod[440nm]',
+               'absorption_aerosol_optical_depth_5': 'absorption_aod[675nm]',
+               'absorption_aerosol_optical_depth_6': 'absorption_aod[870nm]',
+               'absorption_aerosol_optical_depth_7': 'absorption_aod[1020nm]'}
+
+    # Define varDict variables
+    print('Define varDict variables')
+    for key, value in obsvars.items():
+        print(key, value)
+        varDict[key]['valKey'] = key, writer.OvalName()
+        varDict[key]['errKey'] = key, writer.OerrName()
+        varDict[key]['qcKey'] = key, writer.OqcName()
+
+    # Define loc_mdata
+    print('Define loc_mdata')
```

Comment:
do we need all of these debug print statements?

Reply:
@CoryMartin-NOAA Thanks for reviewing the codes. I modified and cleaned up the codes as you suggested except that units['latitude'] = 'degree' is retained.  'degree' is the unit for lon/lat in the IODA name convention document. Besides, because single scattering albedo (SSA) is not directly used in the AAOD assimilation and calculation,  I removed its related variables in the code, files in test/testinput and test/CMakeLists.txt.